packages/core/src/root.ts

```diff
@@ -239,6 +241,16 @@ export function createRemoteRoot<
     const normalizedChild =
       typeof child === 'string' ? remoteRoot.createText(child) : child;
 
+    if (!validateAllowedComponent(normalizedChild, components)) {
```

Comment:
I had not actually considered validating when inserting children. It's probably a good idea to validate then, but I think it's more important to validate in `createComponent`. It gives earlier feedback and, if we were to track components created by a given `Root`, would let us safely avoid validating on insertions.

Reply:
I was looking at `createComponent` before deciding to use append and insert. The problem with `createComponent` is that this function has to return something. In case of unsupported components, what should we do? Should we throw exception to avoid the return value? If so, the whole app won't work anymore. 

With the approach in this PR, the app still works (at least no crash), other supported components still run. The host has option to handle unsupported components in the UI to let partner / merchant know. 

We can discuss tomorrow on a call.

Reply:
Is it possible to return a stub component or just log an error and move on?

Reply:
Can we not return null?

Reply:
The `perform` call below returns void. So, an early return just works.